### ISSUE TITLE
Decrypting Session Data Using Python

### ISSUE THREAD

**Issue by dev-aakash1998** [bungle/lua-resty-session#129]
Hi, I am using lua-resty-session for managing Keycloak Session from Nginx. I was planning to use the 'default' strategy _(i.e AES256 with SHA512 and HMAC_SHA1)_ for encryption of user data to send it to my Django client. However I am unable to decrypt it from Python end. As far as I understand, the session decryption works like this:

1. Obtaining hkey: 
     a: key = table.concat **{ session_id _(base64)_ + expires _(int)_}** 
     b: hkey = hmac_sha1 **{session_secret _(str)_, key _(string?)_}**

2. Create Cipher from hkey and first 8 bytes of session_id

3. Decrypt Data using the cipher

However, in STEP 1.a, in Python I am unable to concatenate a byte object with a number or a string. Hence I am unable to get the correct **_key_**. Which further means I am unable to obtain the correct _**hkey**_, _**cipher**_ or _**decrypted data**_. 

Please let me know if there's a workaround or if I am missing something. 
Also, if this is not a recommended implementation of decrypting the session data, I'd be open to hearing any alternate strategies to decrypt from the server end (for eg: implementing a sidecar which uses lua to decrypt). 

Thanks in advance!

**Comment by PalAditya** [comment]
@dev-aakash1998 - Were you able to find any solution for this?

**Comment by bungle** [comment]
4.0.0 documents the encryption process here:
https://github.com/bungle/lua-resty-session#data-encryption

The decryption process is opposite.

**Comment by souravs17031999** [comment]
@dev-aakash1998 can you summarize the final decryption process which worked for you as I am also not able to wrap my head around this process just by reading it from readme.

**Comment by souravs17031999** [comment]
@bungle I am decrypting this cookie based session data in Rust (server side), will there be any change in decryption process as mentioned or will I be able to do it as per mentioned docs ?
